internal/pkg/discovery/hello_protocol.go

```diff
@@ -112,36 +156,88 @@ func (h *HelloProtocolHandler) getOurHelloMessage() (*HelloMessage, error) {
 	if err != nil {
 		return nil, err
 	}
+	weight, err := heaviest.ParentWeight()
+	if err != nil {
+		return nil, err
+	}
 
 	return &HelloMessage{
 		GenesisHash:          h.genesis,
 		HeaviestTipSetCids:   heaviest.Key(),
 		HeaviestTipSetHeight: height,
+		HeaviestTipSetWeight: weight,
 	}, nil
 }
 
-func (h *HelloProtocolHandler) receiveHello(ctx context.Context, p peer.ID) (*HelloMessage, error) {
-	s, err := h.host.NewStream(ctx, p, helloProtocolID(h.networkName))
+func (h *HelloProtocolHandler) receiveHello(ctx context.Context, s net.Stream) (*HelloMessage, error) {
+	var hello HelloMessage
+	// Read cbor bytes from stream into hello message
+	// Dragons: this handles ~9 head cids.  We should reduce assumptions
+	buf := make([]byte, 512)
```

Comment:
Please factor out the 512 to a named constant with this note, and remove "dragons". Then either:
- make it bigger, and/or
- file an issue noting this

We don't need dragons for just "we could do this better".

Reply:
@anorth Thanks for feedback!  Made it waaay better by just reading streamed cbor directly upon @Kubuxu's suggestion.  You might want to to take a quick look at latest commit to see diff